### ISSUE TITLE
Harmony Hub disconnects randomly leaving me in a dark room ...

### ISSUE THREAD

**Issue by mediaDS** [iobroker-community-adapters/ioBroker.harmony#67]
Hi
my Harmony Hub disconnects randomly, mostly after midnight. Normally I am watchichg TV at that time - so after a few seconds all devices are shutdown (PowerOff) leaving me flabbergasted. I can see how the LED is turning red on the Hub, then green again. Shortly after that the devices like TV set, Soundbar, Kodi Mediaplayer, Lights and so on are shut down. This happens at least once a day. Is there a way to prevent Shutdown?

Log (mediaDS is the name of my Hub):

```
2020-12-28 01:06:14.539 - info: harmony.0 (18337) [CONNECT] lost Connection to mediaDS (10.42.0.20)
2020-12-28 01:06:31.560 - info: harmony.0 (18337) [CONNECT] Connected to mediaDS (10.42.0.20)
2020-12-28 01:06:32.791 - info: kodi.0 (2498) sending in KODI: System.Shutdown - []
2020-12-28 01:06:32.811 - info: harmony.0 (18337) hub busy, stateChange delayed: harmony.0.mediaDS.TV.PowerOff = 1
2020-12-28 01:07:13.275 - info: mqtt.0 (6044) Client [Helios] connection closed: timeout
2020-12-28 01:07:32.318 - info: harmony.0 (18337) hub busy, stateChange delayed: harmony.0.mediaDS.TV.PowerOff = 1
2020-12-28 01:08:14.430 - info: mqtt.0 (6044) Client [Helios] connected with secret 1609114094422_7988
2020-12-28 01:08:14.460 - info: mqtt.0 (6044) Client [Helios] subscribes on "mqtt.0.Helios.R"

```

Platform linux
Betriebssystem linux
Architektur arm64
CPUs 4
Geschwindigkeit 1296 MHz
Modell Odroid
RAM 1.94 GB
Node.js v12.20.0
NPM 6.14.8

Adapter version 1.2.2
JS Script Engine 4.10.8
Kodi Media Control 2.0.4

**Comment by mediaDS** [comment]
No answer for a month?  Is this adapter dead? 

The same happened again today, log:


```
2021-01-30 22:48:58.419 - info: harmony.0 (1444) [CONNECT] lost Connection to mediaDS (10.42.0.20)
--
2021-01-30 22:49:15.330 - info: harmony.0 (1444) [CONNECT] Connected to mediaDS (10.42.0.20)
2021-01-30 22:49:16.698 - info: harmony.0 (1444) hub busy, stateChange delayed: harmony.0.mediaDS.TV.PowerOff = 1
2021-01-30 22:49:16.695 - info: kodi.0 (15628) sending in KODI: Player.Stop - {"playerid":1}
2021-01-30 22:49:16.714 - info: kodi.0 (15628) sending in KODI: System.Shutdown - []

```

All the sudden everything got switched off while watching a movie.

My Sysinfo:

System: ioBroker
Hostname: Casper
Platform: Linux arm64 @  Odroid-C2 
Memory: 2 GB
Node.js: 12.20.1
NPM: 6.14.10
JS-Controller: 3.2.12
Admin Adpater: 4.2.1
JS Script Engine: 4.10.14
Harmony Adapter: 1.2.2 (from 2019!)

**Comment by foxriver76** [comment]
The adapter does not send any power off command on connection. The shutdown command also seems to come from your Kodi adapter. Maybe you have a script which is triggered on a harmony state and then triggers a Kodi shutdown? ;-)

**Comment by mediaDS** [comment]
No, I don't have such a script. But I do have a script to shutdown everything by pressing the PowerOff button on the Harmony remote. Of course I did not press that :)

Anyway, since this did not happen for weeks now; I make the brave assumption it might have been the hub itself triggering the shutdown. I observed a red LED on the hub just seconds before the shutdown actually happening. Maybe an update or a bug in the Logitech software, I don't know.

I'll update if it happens again (crossing my fingers and toes that it won't).

Thanks,
Dieter
